configs/typescript-react/tsconfig.json

```diff
@@ -1,7 +1,8 @@
 {
-	"extends": "@canonical/typescript-config-base",
+	"extends": "@canonical/typescript-config-node",
 	"compilerOptions": {
-		"lib": ["ES2023", "dom"],
+		"module": "ESNext",
+		"moduleResolution": "Bundler",
```

Comment:
The React config has been further modified to use `ESNext` and `Bundler` for `module` and `moduleResolution`, respectively, to address the following error that occurs when building the UI library:

```
 ../../node_modules/@joshwooding/vite-plugin-react-docgen-typescript/dist/index.d.ts(1,10): error TS2305: Module '"vite"' has no exported member 'Plugin'.
      ../../node_modules/@storybook/addon-themes/dist/index.d.ts(1,59): error TS1479: The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("storybook/internal/types")' call instead.
```

This lines up with the [Typescript docs](https://www.typescriptlang.org/tsconfig/#module) which state that `ESNext` should be used for code that is to be bundled.